### DISCUSSION TITLE
What determines the direction of bridging?

### DISCUSSION THREAD

**Discussion by PhilBaz** [smartavionics/Cura#220] [Q&A]
I have two pieces, each is getting bridging in a different direction. What determines the bridging direction?

For me I preferer longest so there are fewer bridging lines, and thus fewer chances for a missed connection at the edge.

![fhgfhgfh](https://user-images.githubusercontent.com/21558241/221958507-66b7b849-feb5-464b-8387-dfc395bcf239.jpg)

**Comment by smartavionics** [comment]
Hi, can you provide a project file so I can understand what's happening here? Thanks.

**Reply by PhilBaz** [reply]
Ahhh, F. I saved over the project after making some changes and now its putting them both across the short edge. 

Sorry, evidence has been destroyed. If it comes up again Ill send a file. 

If you happen to be digging into it, having the option for shortest edge or longest edge would be great. I find what's most detrimental is when the connection to the inner wall doesn't catch and curls up a bit. Thus the orientation with the fewest inner wall connections will minimize the chance for this issue..... I think.

The little missed connections curl up and that can snowball into surface artifacts.  Blobs, burns.

**Reply by smartavionics** [reply]
So the current code determines the direction of the bridge skin lines thus:

If any of the edges of the skin region are supported, the lines will be orientated so they are parallel to the longest unsupported skin edge. This makes sense for the classic bridge between two supported ends.

If no edges of the skin region are supported, the lines with be orientated so they are 90 deg from the longest skin edge.

Have you tried increasing the bridge skin overlap?
